projects/laji/src/app/shared/service/form.service.ts

```diff
@@ -37,48 +39,50 @@ export class FormService {
     private http: HttpClient,
     private userService: UserService,
     private translate: TranslateService
-  ) {}
+  ) {
+    this.translate.onLangChange.subscribe(this.setLang);
+  }
 
   static isTmpId(id: string): boolean {
     return id?.indexOf(FormService.tmpNs + ':') === 0;
   }
 
-  getForm(formId: string, lang = this.translate.currentLang): Observable<Form.SchemaForm | null> {
+  getForm(formId: string): Observable<Form.SchemaForm | null> {
     if (!formId) {
       return ObservableOf(null);
     }
-    this.setLang(lang);
-    if (!this.formCache[formId]) {
-      this.formCache[formId] = this.lajiApi.get(LajiApi.Endpoints.forms, formId, {lang}).pipe(
+    const cacheKey = getCacheKey(formId, this.translate.currentLang);
+    if (!this.formCache[cacheKey]) {
+      this.formCache[cacheKey] = this.lajiApi.get(LajiApi.Endpoints.forms, formId, {lang: this.translate.currentLang}).pipe(
         catchError(error => error.status === 404 ? ObservableOf(null) : observableThrowError(error)),
         retryWhen(errors => errors.pipe(delay(1000), take(2), concat(observableThrowError(errors)))),
         shareReplay(1)
       );
     }
-    return this.formCache[formId];
+    return this.formCache[cacheKey];
   }
 
-  getFormInJSONFormat(formId: string, lang: string): Observable<Form.JsonForm> {
+  getFormInJSONFormat(formId: string): Observable<Form.JsonForm> {
     if (!formId) {
       return ObservableOf({} as Form.JsonForm);
     }
-    this.setLang(lang);
-    if (!this.jsonFormCache[formId]) {
-      this.jsonFormCache[formId] = this.lajiApi.get(LajiApi.Endpoints.forms, formId, {lang, format: 'json'}).pipe(
+    const lang = this.translate.currentLang;
+    const cacheKey = getCacheKey(formId, lang, 'json');
```

Comment:
The format parameter seems redundant since it already uses different caches for different forms (`formCache` and `jsonFormCache`)